### ISSUE TITLE
`psalm-` and `phpstan-` phpdoc-prefix handling

### ISSUE THREAD

**Issue by boesing** [CuyZ/Valinor#118]
Hey there,

we still have some legacy code out there which was not yet migrated to PHP 7.4 and thus, contains the following code:

```php
final class WhateverResponse
{
     /**
      * @var string
      * @psalm-var non-empty-string
      */
     public $nonEmptyString;

     /**
      * @param string $nonEmptyString
      * @psalm-param non-empty-string $nonEmptyString
      */
     public function __construct($nonEmptyString)
     {
     }
}
```

The reason for this is, that we wanted to provide native types for the IDE while also providing psalm-specific types for static analysis.

This also works with `@phpstan-var`, `@phpstan-param`.

So PHPStorm did not properly supported `non-empty-string` for example (it always showed arguments passed to those methods as "invalid") and therefore, we used this way to narrow the type for static analysis while keeping native types for PHPStorm.

I recently tried to use valinor to map a response to that object and it was not throwing a mapping error due to the fact that the `NonEmptyString`-Type was not detected.

----

I haven't had a deeper look yet, but I guess it would be quite easy to support both `@psalm-` and `@phpstan-` prefixes while parsing docblocks?

Thoughts?

**Comment by romm** [comment]
Hi @boesing, that would make sense to support it indeed, and besides it is already supported for template annotations, see:

https://github.com/CuyZ/Valinor/blob/34f519b5837b1d1e7e2f4f270b4338ad1553eb6d/src/Type/Parser/Template/BasicTemplateParser.php#L19-L26

The change would not be that hard. The `@phpstan/psalm-var/param` annotation should override any basic `@var/param` annotation, so it would need to be well tested. Do you think you could handle it?

The entry point would be there:

https://github.com/CuyZ/Valinor/blob/34f519b5837b1d1e7e2f4f270b4338ad1553eb6d/src/Utility/Reflection/Reflection.php#L109-L124

**Comment by boesing** [comment]
Closed with #120
